pkg/controllers/management/k3sbasedupgrade/deployPlans.go

```diff
@@ -55,7 +55,7 @@ func (h *handler) deployPlans(cluster *v3.Cluster, isK3s, isRke2 bool) error {
 	if err != nil {
 		return err
 	}
-	planClient := planConfig.Plans(metav1.NamespaceAll)
+	planClient := planConfig.Plans("cattle-system")
```

Comment:
The original intent of this seems to have been to disable all plans in all namespaces. I agree that is seems weird that this may have been necessary, but I don't want to second-guess that choice.

Could another possible solution be to get all upgrade plans, but do something like `planConfig.Plans(plan.Namespace).Update(context.TODO(), &plan, metav1.UpdateOptions{})` when updating them on line 76? This is the way the `planClient` is changed on line 128, for example.

Perhaps, even, this line could be removed and line 60 could be updated similarly to get all plans.

Reply:
Yeah that works. I think you're right, the original intent is lost here and someone could be relying on this behavior that we aren't aware of.

Reply:
Yeah that works. I think you're right, the original intent is lost here and someone could be relying on this behavior that we aren't aware of.